### PR TITLE
Performant buffering function

### PR DESCRIPTION
**Pull request by Paalon**
Change `_fill` and `__fill` to better implementation and rename to `buffer!`.
Two functions are unified.

### CODE REVIEW

File: src/buffered_input.jl [hunk 1]
```diff
@@ -15,29 +15,26 @@ mutable struct BufferedInput
...
 
-
-# Read and buffer n more characters
-function __fill(bi::BufferedInput, bi_input::IO, n::Integer)
-    for _ in 1:n
-        c = eof(bi_input) ? '\0' : read(bi_input, Char)
-        i = bi.offset + bi.avail + 1
+# Read and buffer `n` more characters
+function buffer!(bi::BufferedInput, n::Integer)::Nothing
+    for i in bi.offset + bi.avail .+ (1:n)
```
Comment (kescobo) [inline src/buffered_input.jl:20]: Does this amount to `(bi.offset + bi.avail) .+ (1:n)`? If so - can we add those parens to make it explicit? This mix of scalar addition + broadcast seems a little weird.
```

Comment (Paalon) [reply src/buffered_input.jl:20]: Ok. It is not must but I make it explicit.

**Review by GunnarFarneback — APPROVED** [review]

**Comment by kescobo** [comment]
ready to merge after conflict resolution

**Comment by Paalon** [comment]
All change of this PR was already merged into master from other PRs, so I close.

**Pull request closed without merging**